rust/src/lib/nm/nm_dbus/connection/ip.rs

```diff
@@ -272,6 +298,21 @@ impl ToDbusValue for NmSettingIp {
     }
 }
 
+impl NmSettingIp {
+    fn i32_to_replace_local_rule(val: Option<i32>) -> Option<bool> {
```

Comment:
Why not make the return value as `Result<Option<bool> , Nmerror>` ?

Reply:
I don't think so. Here we are reading the NmSettingIp, in NM this property is Ternanry which means the valid values are -1, 0, 1. It is very unlikely that we are going to get something different and if that happens we should not fail to the user.

The user will get confused which such strange error, instead we just raise a warning so if this happens and we need to debug we can notice it quickly.